<a href="https://colab.research.google.com/github/anirudhawagh/Technohack_Task2/blob/main/Technohacktask2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name** -  Calculate Summary Statistics

# **Problem Statement** - calculate summary statistics(mean , median , mode, standard derivation ) for a dataset

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import mode


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Load Dataset
database ='/content/drive/MyDrive/Task2 dataset.csv'
df =pd.read_csv(database)

In [ ]:
#First Look
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [ ]:
#First 5rows of dataset
df. head()


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
# Identify categorical columns
categorical_columns = df.select_dtypes(include=['object', 'category']).columns.tolist()
print("Categorical columns:")
print(categorical_columns)


Categorical columns:
['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


In [ ]:
# Identify numerical columns
numerical_columns = df.select_dtypes(include=['int64', 'float64']).columns.tolist()
print("Numerical columns:")
print(numerical_columns)

Numerical columns:
['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']


In [ ]:
# Select numerical columns
numerical_columns = ['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']

# Calculate mean, median, mode, and standard deviation for each numerical column
summary_statistics = pd.DataFrame({
   # 'Column': numerical_columns,
    'Mean': df[numerical_columns].mean(),
    'Median': df[numerical_columns].median(),
    'Mode': [mode(df[col])[0] for col in numerical_columns],
    'Standard Deviation': df[numerical_columns].std()
})

# Display the summary statistics table
summary_statistics


Mean    Median  Mode  Standard Deviation
PassengerId  446.000000  446.0000  1.00          257.353842
Survived       0.383838    0.0000  0.00            0.486592
Pclass         2.308642    3.0000  3.00            0.836071
Age           29.699118   28.0000   NaN           14.526497
SibSp          0.523008    0.0000  0.00            1.102743
Parch          0.381594    0.0000  0.00            0.806057
Fare          32.204208   14.4542  8.05           49.693429

In [ ]:
# Calculate summary statistics for 'PassengerId' grouped by 'Sex'
passenger_id_statistics = df.groupby('Sex')['PassengerId'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
passenger_id_statistics = passenger_id_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_passenger_id_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': passenger_id_statistics['mean'],
    'Median': passenger_id_statistics['median'],
    'Mode': passenger_id_statistics['Mode'],
    'Standard Deviation': passenger_id_statistics['std']
})

# Add a line indicating that this table represents 'PassengerId' statistics
print("PassengerId Statistics:")
# Display the formatted summary statistics table for 'PassengerId'
print(formatted_passenger_id_statistics)


PassengerId Statistics:
      Sex        Mean  Median  Mode  Standard Deviation
0  Female  431.028662   414.5     2          256.846324
1    Male  454.147314   464.0     1          257.486139


In [ ]:
# Calculate summary statistics for 'Survived' grouped by 'Sex'
survived_statistics = df.groupby('Sex')['Survived'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
survived_statistics = survived_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_survived_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': survived_statistics['mean'],
    'Median': survived_statistics['median'],
    'Mode': survived_statistics['Mode'],
    'Standard Deviation': survived_statistics['std']
})

# Add a line indicating that this table represents 'Survived' statistics
print("Survived Statistics:")
# Display the formatted summary statistics table for 'Survived'
print(formatted_survived_statistics)


Survived Statistics:
      Sex      Mean  Median  Mode  Standard Deviation
0  Female  0.742038     1.0     1            0.438211
1    Male  0.188908     0.0     0            0.391775


In [ ]:
# Calculate summary statistics for 'Pclass' grouped by 'Sex'
Pclass_statistics = df.groupby('Sex')['Pclass'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
Pclass_statistics = Pclass_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_Pclass_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': Pclass_statistics['mean'],
    'Median': Pclass_statistics['median'],
    'Mode': Pclass_statistics['Mode'],
    'Standard Deviation': Pclass_statistics['std']
})

# Add a line indicating that this table represents 'Pclass' statistics
print("Pclass Statistics:")
# Display the formatted summary statistics table for 'Pclass'
print(formatted_Pclass_statistics)


Pclass Statistics:
      Sex      Mean  Median  Mode  Standard Deviation
0  Female  2.159236     2.0     3             0.85729
1    Male  2.389948     3.0     3             0.81358


In [ ]:
# Calculate summary statistics for 'Age' grouped by 'Sex'
age_statistics = df.groupby('Sex')['Age'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
age_statistics = age_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_age_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': age_statistics['mean'],
    'Median': age_statistics['median'],
    'Mode': age_statistics['Mode'],
    'Standard Deviation': age_statistics['std']
})

# Add a line indicating that this table represents 'Age' statistics
print("Age Statistics:")
# Display the formatted summary statistics table for 'Age'
print(formatted_age_statistics)


Age Statistics:
      Sex       Mean  Median  Mode  Standard Deviation
0  Female  27.915709    27.0   NaN           14.110146
1    Male  30.726645    29.0   NaN           14.678201


In [ ]:
# Calculate summary statistics for 'SibSp' grouped by 'Sex'
sibsp_statistics = df.groupby('Sex')['SibSp'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
sibsp_statistics = sibsp_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_sibsp_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': sibsp_statistics['mean'],
    'Median': sibsp_statistics['median'],
    'Mode': sibsp_statistics['Mode'],
    'Standard Deviation': sibsp_statistics['std']
})

# Add a line indicating that this table represents 'SibSp' statistics
print("SibSp Statistics:")
# Display the formatted summary statistics table for 'SibSp'
print(formatted_sibsp_statistics)


SibSp Statistics:
      Sex      Mean  Median  Mode  Standard Deviation
0  Female  0.694268     0.0     0            1.156520
1    Male  0.429809     0.0     0            1.061811


In [ ]:
# Calculate summary statistics for 'Parch' grouped by 'Sex'
parch_statistics = df.groupby('Sex')['Parch'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
parch_statistics = parch_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_parch_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': parch_statistics['mean'],
    'Median': parch_statistics['median'],
    'Mode': parch_statistics['Mode'],
    'Standard Deviation': parch_statistics['std']
})

# Add a line indicating that this table represents 'Parch' statistics
print("Parch Statistics:")
# Display the formatted summary statistics table for 'Parch'
print(formatted_parch_statistics)


Parch Statistics:
      Sex      Mean  Median  Mode  Standard Deviation
0  Female  0.649682     0.0     0            1.022846
1    Male  0.235702     0.0     0            0.612294


In [ ]:
# Calculate summary statistics for 'Fare' grouped by 'Sex'
fare_statistics = df.groupby('Sex')['Fare'].agg(['mean', 'median', lambda x: mode(x)[0], 'std']).reset_index()

# Rename the lambda function column to 'Mode'
fare_statistics = fare_statistics.rename(columns={'<lambda_0>': 'Mode'})

# Create a new DataFrame to format the table
formatted_fare_statistics = pd.DataFrame({
    'Sex': ['Female', 'Male'],
    'Mean': fare_statistics['mean'],
    'Median': fare_statistics['median'],
    'Mode': fare_statistics['Mode'],
    'Standard Deviation': fare_statistics['std']
})

# Add a line indicating that this table represents 'Fare' statistics
print("Fare Statistics:")
# Display the formatted summary statistics table for 'Fare'
print(formatted_fare_statistics)


Fare Statistics:
      Sex       Mean  Median  Mode  Standard Deviation
0  Female  44.479818    23.0  7.75           57.997698
1    Male  25.523893    10.5  8.05           43.138263


In [35]:
# Display the summary statistics table
summary_statistics

Statistic     Sex        Mean  Median  Mode  Standard Deviation
0  PassengerId  female  431.028662   414.5  2.00          256.846324
1  PassengerId    male  454.147314   464.0  1.00          257.486139
0     Survived  female    0.742038     1.0  1.00            0.438211
1     Survived    male    0.188908     0.0  0.00            0.391775
0       Pclass  female    2.159236     2.0  3.00            0.857290
1       Pclass    male    2.389948     3.0  3.00            0.813580
0          Age  female   27.915709    27.0   NaN           14.110146
1          Age    male   30.726645    29.0   NaN           14.678201
0        SibSp  female    0.694268     0.0  0.00            1.156520
1        SibSp    male    0.429809     0.0  0.00            1.061811
0        Parch  female    0.649682     0.0  0.00            1.022846
1        Parch    male    0.235702     0.0  0.00            0.612294
0         Fare  female   44.479818    23.0  7.75           57.997698
1         Fare    male   25.523893    10.5  8.05           43.138263


**Conclusion:**
In this project, I conducted a comprehensive analysis of a dataset containing passenger information, focusing on various summary statistics for different numerical attributes grouped by gender ('Female' and 'Male').

Survival Rates: 'Female' passengers had a much higher survival rate (74%) compared to 'Male' passengers (19%), reflecting the historical practice of prioritizing women and children during emergencies.

Passenger Class: On average, 'Female' passengers occupied slightly lower passenger classes, but both genders predominantly traveled in third class.

Age: 'Female' passengers were, on average, slightly younger than 'Male' passengers, although some age data was missing for both genders.

Siblings/Spouses and Parents/Children: 'Female' passengers, on average, had more siblings/spouses and parents/children aboard.

Fare: 'Female' passengers paid a higher average fare than 'Male' passengers.